### ISSUE TITLE
MacOSX build doesn't work with default Python 3.4 distribution

### ISSUE THREAD

**Issue by nfalco79** [mrj0/jep#19]
Under MacOSX I clone the jep repository.
Download and install package for Mac OSX 32/64bit from python web site.
Download and install Oracle java 7 from oracle site.
When I run python3 setup.py build it fails with
```
$ python3 setup.py clean
Traceback (most recent call last):
  File "setup.py", line 100, in <module>
    include_dirs=get_java_include() + ['src/jep', 'build/include'] + numpy_include,
  File "/Users/builder/tmop/jep/commands/java.py", line 64, in get_java_include
    inc = os.path.join(get_java_home(), inc_name)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/posixpath.py", line 89, in join
    "components") from None
TypeError: Can't mix strings and bytes in path components'
```
If you fix java.py script then build breaks again with
> ld: library not found for -lpython3.4m

Fixed python.py script I gain the native lib for mac os :8ball:

**Comment by nfalco79** [comment]
You have a pull request.

**Comment by ndjensen** [comment]
Thanks for reporting and providing a fix.

Do you know what the name of the python3 library is on Mac OS X?  The build is getting -lpython3.4m from python.py get_python_libs().  That would translate to expecting a libpython3.4m.a file.  Is it a problem with the name of that file or just with the directories searched for that file?

**Comment by nfalco79** [comment]
The lib directory of python that comes with its installer is missing in the search paths variable.

**Comment by nfalco79** [comment]
On my system this is the library folder of the official python 3.4 distribution after the installation:

> /Library/Frameworks/Python.framework/Versions/3.4/lib
$ ls -all
libformw.5.dylib
libformw.dylib -> libformw.5.dylib
libmenuw.5.dylib
libmenuw.dylib -> libmenuw.5.dylib
libncursesw.5.dylib
libncursesw.dylib -> libncursesw.5.dylib
libpanelw.5.dylib
libpanelw.dylib -> libpanelw.5.dylib
libpython3.4.dylib -> ../Python
libpython3.4m.dylib -> ../Python
pkgconfig
python3.4

**Comment by ndjensen** [comment]
I merged the pull request and installed python3.4 on OS X to test it.  Is there still a problem?

**Comment by nfalco79** [comment]
No, I close for mistake.
It's just that the person who should close the bug you.

**Comment by ndjensen** [comment]
Closing as fixed in 3.4.
